sdk/platform/http_client/curl/src/curl_client.cpp

```diff
@@ -32,5 +32,117 @@ Response CurlClient::send()
     }
   }
 
-  return Response(200, "OK\n");
+  return this->m_response;
+}
+
+CURLcode CurlClient::Perform()
+{
+  m_firstHeader = true;
+
+  auto settingUp = SetUrl();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  settingUp = SetHeaders();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  settingUp = SetWriteResponse();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  return curl_easy_perform(m_pCurl);
+}
+
+static void ParseAndSetFirstHeader(std::string const& header, Response** response)
+{
+  // set response code, http version and reason phrase (i.e. HTTP/1.1 200 OK)
+  auto start = std::find(header.begin(), header.end(), '/');
+  start++; // skip symbol
+  auto end = std::find(start, header.end(), '.');
+  auto mayorVersion = std::stoi(std::string(start, end));
+
+  start = end + 1; // start of minor version
+  end = std::find(start, header.end(), ' ');
+  auto minorVersion = std::stoi(std::string(start, end));
+
+  start = end + 1; // start of status code
+  end = std::find(start, header.end(), ' ');
+  auto statusCode = std::stoi(std::string(start, end));
+
+  start = end + 1; // start of reason phrase
+  auto reasonPhrase = std::string(start, header.end() - 2); // remove \r and \n from the end
+
+  // allocate the instance of response to heap
+  *response = new Response(mayorVersion, minorVersion, HttpStatusCode(statusCode), reasonPhrase);
```

Comment:
This should `make_unique` probably. Or Response should just be a member of the thing. Either way this makes CurlClient super thread-unsafe, is that OK?

Reply:
I don't have that code in front of me right now but I would assume there is one instance of CurlClient per I/O request. If so, then thread safety is not an issue.

Reply:
yes, it is one response per instance.